### PR TITLE
Fixes and improvements for port forwarding tests

### PR DESCRIPTION
**Pull request by drieseng**
* Support configuration of **PrintMotd** and **AllowTcpForwarding** options.
* Fix and extend public key authentication tests.
* Fix port forwarding tests.
* Add intermittend output test for **ShellStream**.
* Add few tests for `SftpClient.Open(...)`.

### CODE REVIEW

**Comment by IgorMilavec** [comment]
Can you please push updated TestTools? PrintMotd and AllowTcpForwarding are missing in SshdConfig.

**Comment by drieseng** [comment]
> Can you please push updated TestTools? PrintMotd and AllowTcpForwarding are missing in SshdConfig.

Done

File: src/SshNetTests/SshTests.cs [hunk 1]
```diff
@@ -642,7 +661,7 @@ public void Ssh_RemotePortForwarding()
                 using (var s = client.CreateShellStream("a", 80, 25, 800, 600, 200))
                 {
                     s.WriteLine($"telnet {forwardedPort1.BoundHost} {forwardedPort1.BoundPort}");
-                    s.Expect("Escape character is '^]'.");
+                    s.Expect($"Connected to {forwardedPort1.BoundHost}\r\n");
```
Comment (IgorMilavec) [inline src/SshNetTests/SshTests.cs:664]: The tests hang here. Expect() should be called with timeout (5 seconds should do it).

Comment (IgorMilavec) [reply src/SshNetTests/SshTests.cs:664]: We have a different issue open for this, so maybe we should leave this for later.
This PR definitely improved stability, so I think you can merge it.

**Review by IgorMilavec — APPROVED** [review]

**Pull request merged**